core/store/src/trie/state_snapshot.rs

```diff
@@ -314,6 +276,37 @@ impl ShardTries {
         }
     }
 
+    fn delete_current_snapshot(&self) {
+        let StateSnapshotConfig { home_dir, hot_store_path, state_snapshot_subdir, .. } =
+            self.state_snapshot_config();
+
+        // This will delete all existing snapshots from file system. If failed, will retry until success
+        let mut delete_state_snapshots_from_file_system = false;
+        let mut file_system_delete_retries = 0;
+        while !delete_state_snapshots_from_file_system && file_system_delete_retries < 3 {
+            delete_state_snapshots_from_file_system =
+                self.delete_all_state_snapshots(home_dir, hot_store_path, state_snapshot_subdir);
+            file_system_delete_retries += 1;
+        }
+
+        // this will delete the STATE_SNAPSHOT_KEY-value pair from db. If failed, will retry until success
+        let mut delete_state_snapshot_from_db = false;
+        let mut db_delete_retries = 0;
+        while !delete_state_snapshot_from_db && db_delete_retries < 3 {
+            delete_state_snapshot_from_db = match self.set_state_snapshot_hash(None) {
+                Ok(_) => true,
+                Err(err) => {
+                    // This will be retried.
+                    tracing::debug!(target: "state_snapshot", ?err, "Failed to delete the old state snapshot for BlockMisc::STATE_SNAPSHOT_KEY in rocksdb");
+                    false
+                }
+            };
+            db_delete_retries += 1;
+        }
+
```

Comment:
Can you do a check here if both `delete_state_snapshots_from_file_system` and `delete_state_snapshot_from_db` are true and if not print an error message?

Reply:
Changed code to print the error on every failure. We don't want to panic here though